BetterHolidays/markets/holidays.py

```
@@ -115,18 +115,28 @@ class GoodFriday(CommonHoliday):
+            try:
+                elements = html.inner_html.advanced_find("a", attrs={"href": f"https://www.calendar-365.co.uk/holidays/good-friday.html", "class": "link_arrow", "title": "Good Friday 2026", "text": "Good Friday"})
+                if not elements:
+                    raise ValueError(f"Could not find Good Friday information for {year}")
+            except Exception as e:
+                raise ValueError(f"Error finding Good Friday information: {str(e)}")
+
+            tr = elements[0].parent.parent
+            day, month, _ = self.regex.match(tr.children[0].text).groups()
+            return dt.date(year, MONTHS_MAP[month.upper()], int(day))
+        except Exception as e:
+            raise ValueError(f"Error determining Good Friday date for {year}: {str(e)} ({type(e)})")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)